### PR TITLE
BYOC Basics Article

### PR DESCRIPTION
**Pull request by marians**
This PR adds a basic documentation article about BYOC. It should be the entry point for customers interested in BYOC, regardless of which provider and UI they use.

For https://github.com/giantswarm/giantswarm/issues/3987

### CODE REVIEW

File: src/content/basics/byoc/index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+categories = ["basics"]
++++
+
+# Bring Your Own Cloud (BYOC)
+
+The Giant Swarm architecture distinguishes between the control plane and tenant clusters, with the control plane enabling the creation and operation of tenant clusters and tenant clusters running your Kubernetes workloads.
+
+## What BYOC is good for {#benefits}
+
+Both on AWS and Azure, tenant cluster resources exist in an account (or in Azure terms: a subscription) separate from the one hosting the control plane resources. We configure a default account to use for all tenant clusters in an installation. Both accounts, the one for the control plane and the default one for tenant clusters, are in the customer's jurisdiction.
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]: We don't use separate accounts for _every_ installation.

File: src/content/basics/byoc/index.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+- On AWS, Giant Swarms uses two separate IAM roles to assume in order to act in the tenant cluster account: one for use by automation, one for technical support staff. Details on the exact permissions required are found in our guide on [preparing an AWS account to run Giant Swarm tenant clusters](/guides/prepare-aws-account-for-tenant-clusters/).
+
+- On Azure, one service principal is configured for Giant Swarm, used by automation and technical support staff. Detais are found in our guide on [preparing an Azure subscription to run Giant Swarm tenant clusters](/guides/prepare-azure-subscription-for-tenant-clusters/).
+
+## Details you should know about {#details}
+
+- Cloud provider account/subscription credentials are specified on the (Giant Swarm) **organization level**.
+
+- Cloud provider credentials are **immutible**. Once specified on an organization, cloud provider credentials cannot be modified or deleted. In order to switch to new cloud provider credentials you'll have to create a new organization and migrate to new cluster owned by that organization.
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]:
```suggestion
- Cloud provider credentials are **immutable**. Once specified on an organization, cloud provider credentials cannot be modified or deleted. In order to switch to new cloud provider credentials you'll have to create a new organization and migrate to new cluster owned by that organization.
```

File: src/content/basics/byoc/index.md (hunk 1 above)
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]: Not all installations use separate accounts for control plane and tenant clusters - icconsult uses the same account for both, for example

File: src/content/basics/byoc/index.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+- If an organization does not yet have provider credentials configured and alrready has tenant clusters, these clusters are running in the default tenant cluster account. Setting credentials for this organization does not affect the tenant clusters created before.
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]: > Setting credentials for this organization does not affect the tenant clusters created before.

Does
> Setting credentials for this organization does not affect the tenant clusters created already.

Make more sense?

File: src/content/basics/byoc/index.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Get started
+
+To create clusters in a new cloud provider account, first you need to provide the credentials to the organization you'd like to use for this purpose. You are free to create a new organization for this purpose if you like. Organizations can be created in the Giant Swarm web UI, also known as happa, or via the [Giant Swarm API](/api/#operation/addOrganization).
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]:
```suggestion
To create clusters in a new cloud provider account, you first need to provide the credentials to the organization you'd like to use for this purpose. You are free to create a new organization for this purpose if you like. Organizations can be created in the Giant Swarm web UI, also known as happa, or via the [Giant Swarm API](/api/#operation/addOrganization).
```

File: src/content/basics/byoc/index.md (hunk 3 above)
Comment (marians) [reply src/content/basics/byoc/index.md]: Changed

File: src/content/basics/byoc/index.md (hunk 1 above)
Comment (marians) [reply src/content/basics/byoc/index.md]: I added a note to explain this.

**Comment by marians** [comment]
PTAL

**Comment by marians** [comment]
PTAL @JosephSalisbury

File: src/content/basics/byoc/index.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+- On AWS, Giant Swarms uses two separate IAM roles to assume in order to act in the tenant cluster account: one for use by automation, one for technical support staff. Details on the exact permissions required are found in our guide on [preparing an AWS account to run Giant Swarm tenant clusters](/guides/prepare-aws-account-for-tenant-clusters/).
+
+- On Azure, one service principal is configured for Giant Swarm, used by automation and technical support staff. Detais are found in our guide on [preparing an Azure subscription to run Giant Swarm tenant clusters](/guides/prepare-azure-subscription-for-tenant-clusters/).
+
+## Details you should know about {#details}
+
+- Cloud provider account/subscription credentials are specified on the (Giant Swarm) **organization level**.
+
+- Cloud provider credentials are **immutable**. Once specified on an organization, cloud provider credentials cannot be modified or deleted. In order to switch to new cloud provider credentials you'll have to create a new organization and migrate to new cluster owned by that organization.
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]: > you'll have to create a new organization and migrate to new cluster owned by that organization.

'migrate to new clusters owned by that organization'?

File: src/content/basics/byoc/index.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+- If an organization does not yet have provider credentials configured and alrready has tenant clusters, these clusters are running in the default tenant cluster account. Setting credentials for this organization does not affect the tenant clusters created already.
```
Comment (JosephSalisbury) [inline src/content/basics/byoc/index.md]:
```suggestion
- If an organization does not yet have provider credentials configured and already has tenant clusters, these clusters are running in the default tenant cluster account. Setting credentials for this organization does not affect the tenant clusters created already.
```

File: src/content/basics/byoc/index.md (hunk 5 above)
Comment (marians) [reply src/content/basics/byoc/index.md]: What a nie case for a suggestion :)

```suggestion
- Cloud provider credentials are **immutable**. Once specified on an organization, cloud provider credentials cannot be modified or deleted. In order to switch to new cloud provider credentials you'll have to create a new organization and migrate to new clusters owned by that organization.
```

**Comment by marians** [comment]
Thank you very much @JosephSalisbury ! 💐

**Pull request merged**